packages/parser-javascript/src/estree-jsx.ts

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable @typescript-eslint/interface-name-prefix */
+
+// Based on https://github.com/facebook/jsx/blob/master/AST.md
+
+import {
+    Expression,
+    Identifier,
+    Literal,
+    Node as ESTreeNode,
+    SpreadElement
+} from 'estree';
+
+export * from 'estree';
+
+type Omit<Props, T> = Pick<T, Exclude<keyof T, Props>>;
+
+export type Node =
```

Comment:
If we're happy with this file I can try contributing it to DefinitelyTyped as it should be complete. I'll likely re-work the existing `@types/estree` package slightly to export a few private types like `BaseNode` so they can be re-used instead of copied like I had to do here.

Reply:
We should contribute upstream. Let's hope they don't take too much time to approve the PR.